### PR TITLE
Implemented code to search product through barcode scanner 

### PR DESCRIPTION
**Pull request by azkyakhan**

### CODE REVIEW

**Comment by dt2patel** [comment]
Reviewed on my end for UI work. @adityasharma7 please take look at your end

File: src/views/Search.vue [hunk 1]
```diff
@@ -113,7 +129,65 @@ export default defineComponent({
...
+    async checkCameraPermission() {
+
+      const status = await BarcodeScanner.checkPermission({ force: true });
+
+      if (status.granted) {
+        // the user granted permission
+        return true;
+      } else if (status.denied) {
+        // the user has not given permission, showing alert message to enable camera permission in settings
+        this.presentAlertConfirm(translate('No permission'), translate('Please allow camera access in your settings'));
```
Comment (adityasharma7) [inline src/views/Search.vue:157]: We are translating twice, here and in method

File: src/views/Search.vue [hunk 2]
```diff
@@ -113,7 +129,65 @@ export default defineComponent({
...
+      } else {
+        // showing alert if there is any other error
+        this.presentAlertConfirm(translate('Error'), translate('Unable to start camera, please try again'));
```
Comment (adityasharma7) [inline src/views/Search.vue:160]: We are translating twice, here and in method

File: src/store/modules/product/actions.ts [hunk 3]
```diff
@@ -84,11 +83,49 @@ const actions: ActionTree<ProductState, RootState> = {
     commit(types.PRODUCT_ADD_TO_UPLD_PRDTS, { product: payload })
   },
 
-  updateCurrentProduct ({ commit, state }, payload) {
+  async setCurrent ({ commit, state }, payload) {
+    const upc = payload.product ? payload.product.upc : payload.upc;
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:87]: I think we should handle based upon upc or sku in payload

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Comment by adityasharma7** [comment]
Done with https://github.com/hotwax/inventorycount/pull/43. Closing

**Pull request closed without merging**